tests/app/views/test_search.py

```diff
@@ -20,356 +20,89 @@ def find_search_summary(res_data):
+        self._setUp(api_client)
+        assert self.response.status_code == 200
+
+    def test_catalogue_second_page_loads(self, api_client):
+        self._setUp(api_client, '?page=2')
+        assert self.response.status_code == 200
+
+    def test_catalogue_has_filter_checkboxes(self, api_client):
+        self._setUp(api_client)
+        for role_row in self.roles_json['roles']:
+            role = role_row['role'].replace('Senior ', '').replace('Junior ', '')  # Mind the white space after Junior
+            checkbox = self.page.get_element_by_id(role.lower().replace(' ', '-'), None)
+            assert (checkbox is not None and checkbox.name == 'role' and checkbox.type == 'checkbox')
+
+    def test_catalogue_first_page_has_correct_number_of_results(self, api_client):
+        self._setUp(api_client)
+        assert len(self.page.find_class('supplier-result')) == len(self.results_json['hits']['hits'])
+
+    def test_catalogue_second_page_has_back_page_link(self, api_client):
+        self._setUp(api_client, '?page=2')
+        found = False
+        for element, attribute, link, pos in self.page.find_class('pagination')[0].iterlinks():
+            if 'page=1' in link:
+                found = True
+        assert found
+
+    def test_catalogue_second_page_has_next_page_link(self, api_client):
+        self._setUp(api_client, '?page=2')
+        found = False
+        for element, attribute, link, pos in self.page.find_class('pagination')[0].iterlinks():
+            if 'page=3' in link:
+                found = True
+        assert found
+
+    def test_catalogue_pagination_links_have_supplied_query_string(self, api_client):
+        self._setUp(api_client, '?page=2&sort_term=name&sort_order=desc&role=Agile+Coach&role=Delivery+Manager')
+        found = False
+        for element, attribute, link, pos in self.page.find_class('pagination')[0].iterlinks():
+            # Because it can be in any order
+            if 'page=2' in link and 'sort_term=name' in link and 'sort_order=desc' in link and \
+               'role=Agile+Coach' in link and 'role=Delivery+Manager' in link:
+                found = True
+        assert found
+
+    def test_catalogue_search(self, api_client):
+        self._setUp(api_client)
+        supplier_name = self.results_json['hits']['hits'][0]['_source']['name']
+
+        self._setUp(api_client, '?keyword=%s' % supplier_name.replace(' ', '+'))
+
+        found = False
+        for result in self.page.find_class('supplier-result'):
+            for element, attribute, link, pos in result.iterlinks():
+                if element.text_content().strip() == supplier_name:
+                    found = True
+        assert found
+
+    def test_clear_buttons_have_valid_url(self, api_client):
+        self._setUp(api_client)
+        valid = True
+        for button in self.page.find_class('clear-all'):
+            if button.attrib['href'] != '/marketplace/search/suppliers':
```

Comment:
Flask says I can't use url_for on tests. Removed the import and the hardcoded url is now placed in one single variable for easy change if required.